### ISSUE TITLE
`lexical-core` crates prevents kani from building

### ISSUE THREAD

**Issue by ChristopherRabotin** [model-checking/kani#1768]
The [lexical-core](https://crates.io/crates/lexical-core) crate causes kani to fail the build. I don't know where to start debugging this, but I can provide logs of the Github action.

[failed build](https://github.com/nyx-space/hifitime/actions/runs/3216395200/jobs/5258243859)
[successful build](https://github.com/nyx-space/hifitime/actions/runs/3215092857/jobs/5255944850)

I also tried preventing the compilation of the single function that uses this by adding `#[cfg(not(kani))]` before it, but that didn't prevent the building. If there's a better method, I'm all ears.

**Comment by tedinski** [comment]
(Sorry for the huge number of messages in the logs there, that regression will be fixed in the release next week.)

Looks like:

```
   Compiling lexical-parse-float v0.8.5
thread 'rustc' panicked at 'called `Option::unwrap()` on a `None` value', kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs:213:21
stack backtrace:
   3: kani_compiler::codegen_cprover_gotoc::codegen::rvalue::<impl kani_compiler::codegen_cprover_gotoc::context::goto_ctx::GotocCtx>::codegen_rvalue
   4: kani_compiler::codegen_cprover_gotoc::codegen::statement::<impl kani_compiler::codegen_cprover_gotoc::context::goto_ctx::GotocCtx>::codegen_statement
   5: kani_compiler::codegen_cprover_gotoc::codegen::block::<impl kani_compiler::codegen_cprover_gotoc::context::goto_ctx::GotocCtx>::codegen_block
   6: kani_compiler::codegen_cprover_gotoc::codegen::function::<impl kani_compiler::codegen_cprover_gotoc::context::goto_ctx::GotocCtx>::codegen_function
   7: std::thread::local::LocalKey<T>::with
   8: <kani_compiler::codegen_cprover_gotoc::compiler_interface::GotocCodegenBackend as rustc_codegen_ssa::traits::backend::CodegenBackend>::codegen_crate
   9: <rustc_session::session::Session>::time::<alloc::boxed::Box<dyn core::any::Any>, rustc_interface::passes::start_codegen::{closure#0}>
  10: <rustc_interface::passes::QueryContext>::enter::<<rustc_interface::queries::Queries>::ongoing_codegen::{closure#0}::{closure#0}, core::result::Result<alloc::boxed::Box<dyn core::any::Any>, rustc_errors::ErrorGuaranteed>>
  11: <rustc_interface::queries::Queries>::ongoing_codegen
  12: rustc_interface::interface::create_compiler_and_run::<core::result::Result<(), rustc_errors::ErrorGuaranteed>, rustc_driver::run_compiler::{closure#1}>
  13: <scoped_tls::ScopedKey<rustc_span::SessionGlobals>>::set::<rustc_interface::interface::run_compiler<core::result::Result<(), rustc_errors::ErrorGuaranteed>, rustc_driver::run_compiler::{closure#1}>::{closure#0}, core::result::Result<(), rustc_errors::ErrorGuaranteed>>
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

Kani unexpectedly panicked during compilation.
If you are seeing this message, please file an issue here: https://github.com/model-checking/kani/issues/new?labels=bug&template=bug_report.md

[Kani] current codegen item: codegen_function: bigint::StackVec::<62>::new
_RNvMs0_NtCsbw7udg9h8dA_19lexical_parse_float6bigintINtB5_8StackVecKj3e_E3newB7_
[Kani] current codegen location: Loc { file: "/home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/lexical-parse-float-0.8.5/src/bigint.rs", function: None, start_line: 294, start_col: Some(5), end_line: 294, end_col: Some(31) }
error: could not compile `lexical-parse-float`
```

This looks like it's also:

* #1728

**Comment by adpaco-aws** [comment]
Hi @ChristopherRabotin , thanks for the bug report!

I've been able to reproduce the issue on my machine, and I got the following error message (this doesn't show in the logs you linked):
```
   Compiling proc-macro2 v1.0.46
   Compiling memchr v2.5.0
   Compiling unicode-ident v1.0.5
   Compiling static_assertions v1.1.0
   Compiling quote v1.0.21
   Compiling libm v0.2.5
   Compiling autocfg v1.1.0
   Compiling syn v1.0.102
   Compiling serde v1.0.145
   Compiling serde_derive v1.0.145
   Compiling regex-syntax v0.6.27
   Compiling lexical-util v0.8.5
   Compiling num-traits v0.2.15
   Compiling aho-corasick v0.7.19
   Compiling lexical-parse-integer v0.8.6
   Compiling lexical-parse-float v0.8.5
   Compiling lexical-core v0.8.5
thread 'rustc' panicked at 'called `Option::unwrap()` on a `None` value', kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs:213:77
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

Kani unexpectedly panicked during compilation.
If you are seeing this message, please file an issue here: https://github.com/model-checking/kani/issues/new?labels=bug&template=bug_report.md

[Kani] current codegen item: codegen_function: bigint::StackVec::<62>::new
_RNvMs0_NtCsbw7udg9h8dA_19lexical_parse_float6bigintINtB5_8StackVecKj3e_E3newB7_
[Kani] current codegen location: Loc { file: "/home/ubuntu/.cargo/registry/src/github.com-1ecc6299db9ec823/lexical-parse-float-0.8.5/src/bigint.rs", function: None, start_line: 294, start_col: Some(5), end_line: 294, end_col: Some(31) }
error: could not compile `lexical-parse-float`
warning: build failed, waiting for other jobs to finish...
```

This could be another instance of #1728 and #1662, which are due to Kani's handling of const generics.

**Comment by ChristopherRabotin** [comment]
Good to know, I'll follow the other issues so feel free to close this one.

I know little about formal verification, but if I can help in any way, let me know.

**Comment by ChristopherRabotin** [comment]
Hey, it's me again, sorry. It seems like kani is [still tripping up](https://github.com/nyx-space/hifitime/actions/runs/3253714473/jobs/5341226485) on something in lexical-core. I'm not sure what or where: how would you recommend I start debugging this for my issue to be more helpful?
Thanks

**Comment by zhassan-aws** [comment]
Hi @ChristopherRabotin. The fix to this issue has not been released yet; it will be in this Tuesday's 0.13 release. In the meantime, you can [build Kani from source](https://model-checking.github.io/kani/build-from-source.html) to get the fix.
